### PR TITLE
reduce flakiness of memory test

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

This is the error message we see sometimes in CI: 
```python

_______________ test_multiple_trainer_constant_memory_allocated ________________

tmpdir = local('/tmp/pytest-of-AzDevOps_azpcontainer/pytest-0/test_multiple_trainer_constant0')

    @RunIf(min_gpus=1)
    def test_multiple_trainer_constant_memory_allocated(tmpdir):
        """
        This tests ensures calling the trainer several times reset the memory back to 0.
        """
    
        class TestModel(BoringModel):
            def training_step(self, batch, batch_idx):
                loss = super().training_step(batch, batch_idx)
                self.log("train_loss", loss["loss"])
                return loss
    
            def configure_optimizers(self):
                return torch.optim.Adam(self.layer.parameters(), lr=0.1)
    
        class Check(Callback):
            def on_epoch_start(self, trainer, *_):
                assert isinstance(trainer.training_type_plugin.model, DistributedDataParallel)
    
        initial = torch.cuda.memory_allocated(0)
    
        model = TestModel()
        trainer_kwargs = dict(
            default_root_dir=tmpdir,
            fast_dev_run=True,
            gpus=1,
            accelerator="ddp",
            progress_bar_refresh_rate=0,
            callbacks=Check(),
        )
        trainer = Trainer(**trainer_kwargs)
        trainer.fit(model)
    
        assert trainer.training_type_plugin.model is model
        assert list(trainer.optimizers[0].state.values())[0]["exp_avg_sq"].device == torch.device("cpu")
        assert trainer.callback_metrics["train_loss"].device == torch.device("cpu")
    
        # before measuring the memory force release any leftover allocations, including CUDA tensors
        gc.collect()
        memory_1 = torch.cuda.memory_allocated(0)
>       assert memory_1 == initial
E       assert 3072 == 4096
E         +3072
E         -4096

/__w/1/s/tests/trainer/test_trainer.py:1899: AssertionError

```

The test here is flaky when measured memory is **less** than the initial memory. We can't assert equality strictly, it seems.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGMT !

**Review by kaushikb11 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]
